src/python/pants/invalidation/build_invalidator.py

```diff
@@ -93,8 +94,30 @@ def key_for_target(self, target, transitive=False, fingerprint_strategy=None):
 class BuildInvalidator(object):
   """Invalidates build targets based on the SHA1 hash of source files and other inputs."""
 
-  def __init__(self, root):
-    self._root = os.path.join(root, GLOBAL_CACHE_KEY_GEN_VERSION)
+  class Factory(Subsystem):
+    options_scope = 'build-invalidator'
+
```

Comment:
> Is that even a thing?

It isn't... the cache setup subsystem is implicitly used by all tasks, since it is used by the base class.

But even if the cache is completely disabled, the goal of putting this option here is to begin unifying `invalidation` with `caching`: they've been separate in the past, but they shouldn't be in the future. So this also affects tasks that only use `self.invalidated` blocks.